### PR TITLE
Document hidden=until-found

### LINKED ISSUE #20382: The 'hidden' attribute page does not describe the 'until-found' value for this attribute
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

The browser compatibility section mentions the 'until-found' value for this attribute, but no where else on the page is this described or explained. There is no table of options for the possible values of the 'hidden' attribute.

### What did you expect to see?

A table showing the possible values for the 'hidden' attribute, including the 'until-found' value with a description of what it does.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/global_attributes/hidden`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/hidden
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/global_attributes/hidden/index.md
* Last commit: https://github.com/mdn/content/commit/ae92c3a4a3c76b630bfb8d5d771e8efcc024e353
* Document last modified: 2021-11-19T16:02:29.000Z

</details>

### PR DESCRIPTION
**Pull request by wbamberg**
Fixes https://github.com/mdn/content/issues/20382.

Document the `until-found` value for `hidden`.

We've got:

1. some general remarks that are supposed to apply to both states
2. some remarks about _hidden_ state
3. some remarks about _hidden until found_ state
4. an example showing `until-found`

There are some things I'm not really sure about here. In particular, the existing doc says "Hidden elements shouldn't be linked from non-hidden elements" and the current spec says "Elements that are not themselves hidden must not hyperlink to elements that are hidden" with no indication that this is only for _hidden state_. I don't really understand the point of `until-found` revealing the element on fragment navigation if you can't link to a hidden element. I've kept this admonition in the "general remarks" bit, but I'm not sure that is correct.

We will also need a page on `beforematch`, and BCD for that too.

### CODE REVIEW

File: files/en-us/web/html/global_attributes/hidden/index.md [hunk 1]
```diff
@@ -10,27 +10,149 @@ browser-compat: html.global_attributes.hidden
...
+
+The _hidden_ state indicates that the element is not currently relevant to the page, or that it is being used to declare content for reuse by other parts of the page and should not be directly presented to the user. The browser will not render elements that are in the _hidden_ state.
+
+Web browsers may implement the _hidden_ state using `display: none`, in which case the element will not participate in page layout. This also means that changing the value of the CSS {{cssxref("display")}} property on an element in the _hidden_ state will overrides the state. For instance, elements styled `display: block` will be displayed despite the `hidden` attribute's presence.
+
+### The hidden until found state
+
+In the _hidden until found_ state, the element is hidden but its content will be accessible to the browser's "find in page" feature or to fragment navigation. When these features cause a scroll to an element in a _hidden until found_ subtree, the browser will:
+
+- fire a `beforematch` event on the hidden element
```
Comment (estelle) [inline files/en-us/web/html/global_attributes/hidden/index.md:65]: if this is the first occurrence, link to beforematch

**Comment by estelle** [comment]
> There are some things I'm not really sure about here. In particular, the existing doc says "Hidden elements shouldn't be linked from non-hidden elements" and the current spec says "Elements that are not themselves hidden must not hyperlink to elements that are hidden" with no indication that this is only for _hidden state_. I don't really understand the point of `until-found` revealing the element on fragment navigation if you can't link to a hidden element. I've kept this admonition in the "general remarks" bit, but I'm not sure that is correct.

It sounds to me like this is useful for tablist navigation, showing tabs that link to 'hidden until found' tabpanels.

File: files/en-us/web/html/global_attributes/hidden/index.md (hunk 1 above)
Comment (estelle) [reply files/en-us/web/html/global_attributes/hidden/index.md:65]: Oops, there is no before-match page. Should we add a red link or wait until the page exists and remember to re-visit?

File: files/en-us/web/html/global_attributes/hidden/index.md [hunk 2]
```diff
@@ -10,27 +10,149 @@ browser-compat: html.global_attributes.hidden
...
+
+Also, the element needs to be affected by [layout containment](/en-US/docs/Web/CSS/CSS_Containment) in order to be revealed. This means that if the element in the _hidden until found_ state has a `display` value of `none`, `contents`, or `inline`, then the element will not be revealed by find in page or fragment navigation.
+
+## Examples
+
+### Using until-found
+
+In this example we have:
+
+- Three {{HTMLElement("div")}} elements. The first and the third are not hidden, while the second has `hidden=until-found`and `id=until-found-box` attributes.
```
Comment (estelle) [inline files/en-us/web/html/global_attributes/hidden/index.md:84]:
```suggestion
- Three {{HTMLElement("div")}} elements. The first and the third are not hidden, while the second has `hidden="until-found"`and `id="until-found-box"` attributes.
```

**Review by estelle — APPROVED** [review]

File: files/en-us/web/html/global_attributes/hidden/index.md (hunk 1 above)
Comment (wbamberg) [reply files/en-us/web/html/global_attributes/hidden/index.md:65]: Yup, as I said in the description, I need to write a page for `beforematch`, but will come back and add a link when I have done so.

**Pull request merged**